### PR TITLE
Openstack Security Group hardening

### PR DESCRIPTION
**Pull request by drekle**
Fixes:
https://github.com/kubernetes/kops/issues/6510
https://github.com/kubernetes/kops/issues/6511

/hold
I am currently experiencing an issue which seeing "duplicate security group rule exists" responses from openstack which I cannot easily explain.  I will test in another environment.

/sig openstack

### CODE REVIEW

**Comment by zetaab** [comment]
also these egress rules does not matter at all because there are default rules which will allow everything

File: pkg/model/openstackmodel/firewall.go [hunk 1]
```diff
@@ -36,113 +42,458 @@ type FirewallModelBuilder struct {
...
+func (b *FirewallModelBuilder) addHTTPSRules(c *fi.ModelBuilderContext, sgMap map[string]*openstacktasks.SecurityGroup) error {
+
+	masterName := b.SecurityGroupName(kops.InstanceGroupRoleMaster)
+	nodeName := b.SecurityGroupName(kops.InstanceGroupRoleNode)
+	lbSGName := b.Cluster.Spec.MasterPublicName
+	lbSG := sgMap[lbSGName]
+	masterSG := sgMap[masterName]
+	nodeSG := sgMap[nodeName]
+
+	httpsEgress := &openstacktasks.SecurityGroupRule{
```
[the remark this answers is not in the document]
Comment (drekle) [reply pkg/model/openstackmodel/firewall.go]: To allow proxying the the masters and too allow all nodes to communicate with the api server.  However I missed this gem:
```
When a new security group is created, rules to allow all egress traffic are automatically added.
```
https://wiki.openstack.org/wiki/Neutron/SecurityGroups

And maybe should just remove all egress rules.

**Comment by drekle** [comment]
> My instances do have 10.1.0.0/16 cidr, so I added 10.1.0.0/16 https to master SG then it started to work.

Ill open this up to the subnet.

File: pkg/model/openstackmodel/firewall.go [hunk 2]
```diff
@@ -36,113 +42,448 @@ type FirewallModelBuilder struct {
...
+				PortRangeMin:   i(443),
+				PortRangeMax:   i(443),
+				RemoteIPPrefix: s(apiAccess),
+			})
+		}
+		//Allow masters ingress from the sg
+		addDirectionalGroupRule(c, masterSG, lbSG, httpsIngress)
+
+		//FIXME: Octavia port traffic appears to be denied though its port is in lbSG
+		if b.usesOctavia() {
```
Comment (drekle) [inline pkg/model/openstackmodel/firewall.go]: @zetaab I don't have an octavia openstack atm.  Would you be able to see if this fixes your issue?

**Comment by drekle** [comment]
> * why port 443 is open in nodes?

This was unnecessary and I removed this.

> * could you mark all calico ports, at least i do not know some of these openings

@zetaab They should be marked in two places.  The ETCD rule function adds 4001 for the CNI implementations which require communication to etcd.  The CNI rule function for specific communication ports.

**Comment by drekle** [comment]
@zetaab if you are comfortable with this I was able to get around my issues in a local testing environment.

**Comment by zetaab** [comment]
I will test this when I have time, hopefully later today

**Pull request merged**